pokemongo_bot/cell_workers/incubate_eggs_worker.py

```diff
@@ -2,65 +2,110 @@
+                continue
+            if "pokemon_data" in inv_data:
+                pokemon = inv_data.get("pokemon_data", {})
+                if pokemon.get("is_egg", False) and "egg_incubator_id" not in pokemon:
+                    self.eggs.append({"id": pokemon.get("id", -1), "km": pokemon.get("egg_km_walked_target", -1), "used": False})
+                continue
+            if "player_stats" in inv_data:
+                self.km_walked = inv_data.get("player_stats", {}).get("km_walked", 0)
 
+    def _hatch_eggs(self):
+        self.bot.api.get_hatched_eggs()
+        response_dict = self.bot.api.call()
         try:
-            inv = reduce(dict.__getitem__, [
-                "responses", "GET_INVENTORY", "inventory_delta", "inventory_items"], response_dict)
+            result = reduce(dict.__getitem__, ["responses", "GET_HATCHED_EGGS"], response_dict)
         except KeyError:
             pass
         else:
-            for inv_data in inv:
-                inv_data = inv_data.get("inventory_item_data", {})
-
-                if "egg_incubators" in inv_data:
-                    for incubator in inv_data.get("egg_incubators", {}).get("egg_incubator", []):
-                        if "pokemon_id" not in incubator:
-                            incubators.append({"id":incubator.get("id", -1), "used":False})
-
-                if "pokemon_data" in inv_data:
-                    pokemon = inv_data.get("pokemon_data", {})
-                    if pokemon.get("is_egg", False) and "egg_incubator_id" not in pokemon:
-                        eggs.append({"id": pokemon.get("id", -1), "km": pokemon.get("egg_km_walked_target", -1), "used": False})
-
-            sorting = self.config.longer_eggs_first
-            eggs.sort(key=lambda x: x.get("km"), reverse=sorting)
-
-            for incubator in incubators:
-                if incubator["used"]:
-                    continue
-
-                for egg in eggs:
-                    if egg["used"] or egg["km"] == -1:
-                        continue
-
-                    self.api.use_item_egg_incubator(item_id=incubator["id"], pokemon_id=egg["id"])
-                    ret = self.api.call()
-                    if ret:
-                        code = ret.get("responses", {}).get("USE_ITEM_EGG_INCUBATOR", {}).get("result", 0)
-                        if code == 1:
-                            logger.log('Successfully incubated a ' + str(egg["km"]) + "km egg", 'green')
-                            egg["used"] = True
-                            incubator["used"] = True
-                            break
-                        elif code == 5 or code == 7:
-                            incubator["used"] = True
-                            break
-                        elif code == 6:
-                            egg["used"] = True
-
+            if 'pokemon_id' in result:
+                pokemon_ids = [id for id in result['pokemon_id']]
```

Comment:
Unused variable `pokemon_ids`. Maybe you wanted to print a new pokemon, that was hatched from the egg?